diff --git a/games-frame-pacing/CMakeLists.txt b/games-frame-pacing/CMakeLists.txt
--- a/games-frame-pacing/CMakeLists.txt
+++ b/games-frame-pacing/CMakeLists.txt
@@ -121,6 +121,8 @@ target_link_libraries( swappy
                         log
 			atomic)
 
+# Enable support for non-4k virtual page sizes
+target_link_options(swappy PRIVATE -z max-page-size=16384)
 
 option(BUILD_TEST "Build native tests" OFF)
 if(BUILD_TEST)
